shell/browser/native_window_mac.mm

```diff
@@ -492,6 +492,8 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   // When this is a sheet showing, performClose won't work.
   if (is_modal() && parent() && IsVisible()) {
     [parent()->GetNativeWindow().GetNativeNSWindow() endSheet:window_];
+    // Manually emit close event (not triggered from close fn)
+    [window_delegate_ windowShouldClose:window_];
```

Comment:
Using `NotifyWindowCloseButtonClicked()` directly is probably better, more code may be added in `windowShouldClose` in future and it would then cause side effects here.